### PR TITLE
Update ldsUtils.js

### PR DESCRIPTION
**Pull request by sgkmills**
Have the ldsUtils able to reduce the errors from the UI when doing a DML operation and you get fieldErrors.  For example, an error returned from a datatable update

### What does this PR do?
Have the ldsUtils able to reduce the errors from the UI when doing a DML operation and you get fieldErrors.  For example, an error returned from a datatable update

### What issues does this PR fix or reference?
the reduce function currently doesn't display fieldErrors on a DML operation
#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

The loigc in reduceErrors can decode fieldErrors on a DML operation.    Will return all the the message element from all the arrays in 'error.body.output.fieldErrors' object

### CODE REVIEW

**Comment by sgkmills** [comment]
I signed the CLA back on 8/22, but still see the tag CLA missing.  See image below:

image

**Comment by adityanaag3** [comment]
Thanks @sgkmills The CI is failing due to formatting issues.

Please run `npm install` in your project directory, and run `npm run prettier`. This will auto-format the code. You should then re-commit the `ldsUtils.js` file.

Also, please update the Jest tests for this file to test the `fieldErrors` logic.

**Comment by sgkmills** [comment]
Hi,

I'm not too familiar with the workflow for making the updates so I did what I thought was correct.  Please let me know if I am doing it wrong and why I am getting the error.

I cloned the forked copy of the lwc-recipes from GitHub so I can make the changes in VSCode.  I then ran 'npm install' and 'npm run prettier'  Lastly, I updated the Jest tests for this file to test the fieldErrors logic.  Everything works fine in VSCode and the Jest tests I added are passing, but when I try to commit the ldsUtils.js and ldsUtils.test.js changes back to the patch-1 branch, I get the following message: 

image

**Comment by sgkmills** [comment]
> Hi,
> 
> I'm not too familiar with the workflow for making the updates so I did what I thought was correct. Please let me know if I am doing it wrong and why I am getting the error.
> 
> I cloned the forked copy of the lwc-recipes from GitHub so I can make the changes in VSCode. I then ran 'npm install' and 'npm run prettier' Lastly, I updated the Jest tests for this file to test the fieldErrors logic. Everything works fine in VSCode and the Jest tests I added are passing, but when I try to commit the ldsUtils.js and ldsUtils.test.js changes back to the patch-1 branch, I get the following message:
> 
> image

I am running on a Windows machine and I fixed this error by putting #!/bin/sh as the first line of the pre-commit file.

**Comment by sgkmills** [comment]
I did all that was expected and I guess I'm waiting for the maintainer to approve the running workflows.  Is that an automated process or am I supposed to do something?

**Review by adityanaag3 — APPROVED** [review]

**Review by pozil — CHANGES_REQUESTED** [review]
Thanks for the PR, could you address the comment I left?
Also `package-lock.json` should not be changed as you did not modify dependencies so please revert it.

File: force-app/main/default/lwc/ldsUtils/ldsUtils.js
Comment (sgkmills) [reply force-app/main/default/lwc/ldsUtils/ldsUtils.js]: I made the optional chaining change, and also reverted the `package-lock.json` file.  Let me know if everything is correct now. Thanks!

**Review by pozil — APPROVED** [review]
Looking great, thanks for the changes @sgkmills

**Pull request merged**
